File: dotnet/niuz.application/finance/Payment.cs
using System;

namespace niuz.application.finance
{
    public class Payment
    {
        public int Amount { get; }

        public string BankAccount { get; }

        public string Recipient { get; }

        public string Description { get; }

        public Payment(int amount, string bankAccount, string recipient, string description)
        {
            Amount = amount;
            BankAccount = bankAccount;
            Recipient = recipient;
            Description = description;
        }

        protected bool Equals(Payment other)
        {
            return Amount == other.Amount && BankAccount == other.BankAccount && Recipient == other.Recipient && Description == other.Description;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Payment) obj);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Amount, BankAccount, Recipient, Description);
        }
    }
}

File: java/src/main/java/com/dddeurope/niuz/finance/Payment.java
package com.dddeurope.niuz.finance;

import java.util.Objects;

public class Payment {
    private final int amount;
    private final String bankAccount;
    private final String recipientName;
    private final String description;

    public Payment(int amount, String bankAccount, String recipientName, String description) {
        this.amount = amount;
        this.bankAccount = bankAccount;
        this.recipientName = recipientName;
        this.description = description;
    }

    public int getAmount() {
        return amount;
    }

    public String getBankAccount() {
        return bankAccount;
    }

    public String getRecipientName() {
        return recipientName;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        return "Payment{" +
                "amount=" + amount +
                ", bankAccount='" + bankAccount + '\'' +
                ", recipientName='" + recipientName + '\'' +
                ", description='" + description + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Payment payment = (Payment) o;
        return amount == payment.amount && Objects.equals(bankAccount, payment.bankAccount) && Objects.equals(recipientName, payment.recipientName) && Objects.equals(description, payment.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(amount, bankAccount, recipientName, description);
    }
}
